hooks/node_handlers/base_import_handler.py

```diff
@@ -543,6 +843,484 @@ def refresh_file_path_from_version(self, kwargs):
         publish_data["version_number"] = int(resolved_version)
         self._update_publish_data_parm(node, publish_data, version_policy)
 
+    #############################################################################################
+    # Methods for populating from node
+    #############################################################################################
+
+    def _resolve_work_version(self, all_versions, current):
+        """
+        From a given string, resolve the current version.
+        Either a specified version or the next version in the sequence.
+
+        :param list(int) all_versions: All the existing versions.
+        :param str current: The currently selected version option.
+
+        :rtype: int or NoneType
+        """
+        if all_versions:
+            resolved = max(all_versions)
+            if current != self.LATEST_POLICY:
+                current = int(current)
+                if current in all_versions:
+                    resolved = current
+            return resolved
+
+    def _get_template_fields_and_work_versions(self, parent_node, parm_name):
+        """
+        Get the template, template fields and all versions from the file path
+        associated with the given node and parameter name.
+
+        This method will initially use the node handler for the given node to
+        retrieve the template, then falls back to :meth:`sgtk.Sgtk.template_from_path`.
+
+        If no template can be found, all versions will be calculated using regexes
+        that look for `v#` in the file path.
+
+        :param parent_node: The node to retrieve the data from.
+        :param parm_name: The name of the parameter that contains the file path.
+
+        :returns: A :class:`tuple` of :class:`sgtk.Template`, dict and list(int)
+        """
+        all_versions = []
+        template = None
+        fields = {}
+        if parent_node:
+            path_parm = parent_node.parm(parm_name)
+            if path_parm:
+                path = path_parm.evalAsString()
+                node_handler = self.parent.node_handler(parent_node)
+                if node_handler and hasattr(node_handler, "OUTPUT_PARM"):
+                    template = node_handler.get_work_template(parent_node)
+                else:
+                    template = self.sgtk.template_from_path(path)
```

Comment:
You might want to wrap this with a `try...except` in case `path` is a naughty value and `self.sgtk.template_from_path` errors.
```suggestion
                    try:
                        template = self.sgtk.template_from_path(path)
                    except sgtk.TankError:
                        template = None  # or maybe just "pass", but this is more explicit
```